presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java

```diff
@@ -763,11 +763,11 @@ else if (!result.getBlocked().isDone()) {
             }
         }
         catch (Throwable t) {
+            scheduling.set(false);
```

Comment:
Maybe put some of the commit message as a comment here.

Reply:
So basically the query would fail in either case. But with the race condition it might fail with a strange message? :)

Reply:
This is just a dumb mistake i made. I put the `scheduling.set(false)` to `finally` to ensure that the SqlQueryScheduler is not in a limbo state in case of a failure. I don't know why didn't i put it in the `catch` block originally. 

When this code is in the `finally` block there's a real risk that it will reset the `scheduling` flag when the https://github.com/prestodb/presto/blob/a7889c5559a00cedf91567418e33ca0ed0b310bf/presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java#L762 already submitted the next section for scheduling, effectively setting `scheduling` to `false` when there's scheduling in process.